hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeService.java

```diff
@@ -228,8 +229,12 @@ private void expungeHistoricalVersion(
 			RequestDetails theRequestDetails,
 			ResourceHistoryTablePk theNextVersionId,
 			AtomicInteger theRemainingCount) {
-		ResourceHistoryTable version =
-				myResourceHistoryTableDao.findById(theNextVersionId).orElseThrow(IllegalArgumentException::new);
+		ResourceHistoryTable version = myResourceHistoryTableDao
+				.findById(theNextVersionId)
+				.orElseThrow(() -> new IllegalArgumentException(MessageFormat.format(
```

Comment:
I guess we should probably include an error code here - We try to include them on all newly thrown exception messages that might end up visible to an end user. (FYI we use [this page](https://github.com/hapifhir/hapi-fhir/wiki/Bumping-Error-Code) to reserve them)

```suggestion
				.orElseThrow(() -> new IllegalArgumentException(Msg.code(2701) + MessageFormat.format(
```